pkgs/stdenv/generic/setup.sh

```diff
@@ -483,8 +483,9 @@ activatePackage() {
     # the transition, we do include everything in thatcase.
     #
     # TODO(@Ericson2314): Don't special-case native compilation
-    if [[ ( -z "${strictDeps-}" ||  "$hostOffset" -le -1 ) && -d "$pkg/bin" ]]; then
```

Comment:
we may still want `-d $pkg/bin`. IIRC we get tons more in PATH without it, which can subtly break things that don't expect so long of a path

Reply:
and maybe we should do the same for /share

Reply:
Both are taken care of by `addToSearchPath`.

> The removal of ` && -d "$pkg/bin"` is ok, because this optimization is
already performed by `addToSearchPath`.